test/e2e/preempt.go

```diff
@@ -0,0 +1,206 @@
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	schedulingv1beta1 "volcano.sh/volcano/pkg/apis/scheduling/v1beta1"
+)
+
+var _ = Describe("Job E2E Test", func() {
+	It("schedule high priority job without preemption when resource is enough", func() {
+		ctx := initTestContext(options{
+			priorityClasses: map[string]int32{
+				masterPriority: masterPriorityValue,
+				workerPriority: workerPriorityValue,
+			},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot := oneCPU
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot,
+					min: 1,
+					rep: 1,
+				},
+			},
+		}
+
+		job.name = "preemptee"
+		job.pri = workerPriority
+		preempteeJob := createJob(ctx, job)
+		err := waitTasksReady(ctx, preempteeJob, 1)
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "preemptor"
+		job.pri = masterPriority
+		preemptorJob := createJob(ctx, job)
+		err = waitTasksReady(ctx, preempteeJob, 1)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = waitTasksReady(ctx, preemptorJob, 1)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("schedule high priority job with preemption when resource is NOT enough", func() {
+		ctx := initTestContext(options{
+			priorityClasses: map[string]int32{
+				masterPriority: masterPriorityValue,
+				workerPriority: workerPriorityValue,
+			},
+		})
+		defer cleanupTestContext(ctx)
+
+		slot := oneCPU
+		rep := clusterSize(ctx, slot)
+
+		job := &jobSpec{
+			tasks: []taskSpec{
+				{
+					img: defaultNginxImage,
+					req: slot,
+					min: 1,
+					rep: rep,
+				},
+			},
+		}
+
+		job.name = "preemptee"
+		job.pri = workerPriority
+		preempteeJob := createJob(ctx, job)
+		err := waitTasksReady(ctx, preempteeJob, int(rep))
+		Expect(err).NotTo(HaveOccurred())
+
+		job.name = "preemptor"
+		job.pri = masterPriority
+		job.min = rep / 2
+		preemptorJob := createJob(ctx, job)
+		err = waitTasksReady(ctx, preempteeJob, int(rep)/2)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = waitTasksReady(ctx, preemptorJob, int(rep)/2)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("preemption don't work when podgroup is pending", func() {
+		ctx := initTestContext(options{
+			priorityClasses: map[string]int32{
+				masterPriority: masterPriorityValue,
+				workerPriority: workerPriorityValue,
+			},
+		})
+		defer cleanupTestContext(ctx)
+
+		pgName := "pending-pg"
+		pg := &schedulingv1beta1.PodGroup{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: ctx.namespace,
+				Name:      pgName,
+			},
+			Spec: schedulingv1beta1.PodGroupSpec{
+				MinMember:    1,
+				MinResources: &thirtyCPU,
```

Comment:
donot hard coded , should calculate according to the cluster resource

Reply:
Good advice. There are lots of hard coded in e2e cases and we need a adequate review later